---
title: "RADseq"
output: html_document
---

```{r setup, include=FALSE}
library(tidyr)
library(dplyr)
library(ggplot2)

knitr::opts_chunk$set(echo = TRUE)
```

# Assembly

```{bash, eval=F}
# cluster: /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Assemblies/Mms_female.fa

# NCBI: SAMN42554425
```

## From

```{bash, eval=F}
mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/Assemblies/Mms_female.fa  /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Assemblies
```

# HiFi reads

```{bash, eval=F}
# cluster: /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/HiFi/Mms_female/Mms_female.hifi.fastq.gz
# (mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/HiFi/Female/Mms_female.hifi.fastq.gz  /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/HiFi/Mms_female/ )
# (mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/HiFi/Female/README.md  /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/HiFi/Mms_female/ )

# NCBI: SRR30231907
```

## 

# WGS

```{bash, eval=F}
/nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA
```

## From

Exist in two copies apparently identical on the nas but without Mms ...

```{bash, eval=F}
du -sh /nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/
#74G	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/
du -sh /nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/*
#114M	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Downloading_NovaSeq.log
#24K	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Downloading_NovaSeq.sh
#28G	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Gga
#24K	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Linking_Illumina.log
#24K	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Linking_Illumina.sh
#22G	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Mfg
#160K	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Mms
#25G	/nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Mun


du -sh /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus
#74G	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus
du -sh /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/*
#115M	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Downloading_NovaSeq.log
#24K	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Downloading_NovaSeq.sh
#27G	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Gga
#24K	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Linking_Illumina.log
#24K	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Linking_Illumina.sh
#22G	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Mfg
#160K	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Mms
#25G	/nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Mun

#It is not OK ... Mms is missing ...
##Let's recover stuff from a mix of origins ...
##Gga - OK
cp /nas/FAC/FBM/DEE/tschwand/myrmecophilus/D2c/SR/WGS/Gga/NovaSeq/* /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Gga #(biggest one)
##Mfg - OK
cp /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Mfg/NovaSeq/* /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Mfg #(the one used)

##Mms - OK
cp /work/FAC/FBM/DEE/tschwand/myrmecophilus/WGS/Reads/Mms/Raw/*fastq.gz /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Mms 
##Mun - OK
cp /work/FAC/FBM/DEE/tschwand/myrmecophilus/WGS/Reads/Mun/Raw/*fastq.gz /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Mun #(the one used)

mkdir /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS
mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Gga /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS

#OK
mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/M* /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS
mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/NCBI_10122024.log /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS
mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/NCBI_10122024.sh /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS

cp /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Downloading_NovaSeq.log /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS
cp /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Downloading_NovaSeq.sh /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS
cp /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Linking_Illumina.log /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS
cp /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus/Linking_Illumina.sh /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS

 mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/WGS  /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/DNA/
```

## To NCBi

```{bash, eval=F}
mamba create -p /scratch/vmerel/envs/aspera
mamba activate /scratch/vmerel/envs/aspera
mamba install install rpetit3::aspera-connect

#download keyfile
mkdir -p /users/vmerel/keys
cd /users/vmerel/keys
#manual -> aspera_WGS.key

echo '#!/usr/bin/env bash

date

source ~/.bashrc
conda activate /scratch/vmerel/envs/aspera

mkdir /scratch/vmerel/ncbi_tmp_WGS
cd /scratch/vmerel/ncbi_tmp_WGS
cp /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Gga/*fastq.gz ./
cp /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Mfg/*fastq.gz ./
cp /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Mms/*fastq.gz ./
cp /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/Mun/*fastq.gz ./

ascp -i /users/vmerel/keys/aspera_WGS.key -QT -l100m -k1 -d /scratch/vmerel/ncbi_tmp_WGS/ subasp@upload.ncbi.nlm.nih.gov:uploads/merel.vincent_gmail.com_Rtsqk1nS

date
' > /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/NCBI_10122024.sh

nohup sh /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/NCBI_10122024.sh &> /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/NCBI_10122024.log &

#log moved to WGS see above 
```

du -sh 
mkdir -p /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/
cp -r /nas/FAC/FBM/DEE/tschwand/default/LTS/Myrmecophilus 



# RADseq

NCBI BioSample attributes

```{r,eval=F}

#From the sup. directly
Samples <- read.delim("~/Papers_git/rnx/Samples.csv")
head(Samples)
#You want to keep only the ones with RADseq data, and that were not filter out
Samples <- Samples %>%
  filter(radseq_seq=="TRUE")

#19.0274a should be "unknown", it is the weird gallicus
Samples$radseq_id[Samples$sample=="19.0274a"]<-"ambiguous"
#19.0248a should be "fuscus"
Samples$radseq_id[Samples$sample=="19.0248a"]<-"fuscus"

#isolate to status
Samples$status[Samples$status=="femelle"]="Female"
Samples$status[Samples$status=="larve femelle"]="Female larvae"
Samples$status[Samples$status=="larve indet"]="Larvae"
Samples$status[Samples$status=="larve male"]="Male larvae"
Samples$status[Samples$status=="male"]="Male"

summary(as.factor(Samples$host))
Samples$host[Samples$host=="indet."]="Non determined"

summary(as.factor(Samples$yyyy))
summary(as.factor(Samples$mm))

summary(as.factor(Samples$dep.))
Samples$country = ifelse(Samples$`dep.`=="Grece:", "Greece", "France:")


##For tissue

OldSamples <- read.delim("~/mse/RADseq/Myrmecophilus_samples_simon.csv") %>% select(c("ex_gri","ID"))
colnames(OldSamples) <-c("tissue","sample")
Samples <- left_join(Samples, OldSamples, by="sample")  
summary(as.factor(Samples$tissue))
Samples$tissue[Samples$tissue=="entiere"]="Whole body"
Samples$tissue[Samples$tissue=="femur"]="Leg"


###########################################################################
#You need sample name in isolate because ...
#NCBI: "Your table upload failed because multiple rows in the table (ignoring sample name, sample title and description fields) are identical. Each row must have at least one unique attribute value. Please enter information that distinguishes each sample, eg isolate names or different details of the sample collection. If the distinguishing information is in the sample name, title or description, please copy that value into the appropriate attribute, either one of the predefined attributes or a custom attribute you add. "

df <- data.frame(
  sample_name=Samples$sample,
  organism=
    paste("Myrmecophilus",
          ifelse(Samples$radseq_id %in% c("ambiguous","unknown"), "sp.", Samples$radseq_id)),
  isolate=paste(Samples$sample,Samples$status,sep="-"),
  host=Samples$host,
  collection_date=paste(Samples$yyyy,Samples$mm,sep="-"),
  geo_loc_name=paste(Samples$country,Samples$dep.,sep=" "),
  tissue=Samples$tissue
  
  )

write.table(df,"~/Desktop/ncbi_tmp/tmp.tsv", quote=F, sep="\t", row.names=F)
###########################################################################

```

NCBI SRA metadata

You can't upload raw data and barcodes, NCBI want one filename per sample ...

```{r,eval=F}
Samples <- Samples %>% mutate(filename=paste(substr(Samples$sample,5,9),".fq.gz",sep=""))

df <- data.frame(
  sample_name=Samples$sample,
  library_ID=Samples$sample,
  title="RADseq myrmecophilus",
  library_strategy="RAD-Seq",
  library_source="GENOMIC",
  library_selection="PCR",
  library_layout="single",
  platform="ILLUMINA",
  instrument_model="Illumina HiSeq 2500",
  design_description="double-digest ddRADseq libraries following Brelsford et al. 2017",
  filetype="fastq",
  filename=Samples$filename)


write.table(df,"~/Desktop/ncbi_tmp/tmp_meta.tsv", quote=F, sep="\t", row.names=F)
```

## To NCBI

```{bash, eval=F}
#removing previous tranfer
rm -r /scratch/vmerel/ncbi_tmp_WGS
rm /users/vmerel/keys/aspera_WGS.key

#download keyfile
mkdir -p /users/vmerel/keys
cd /users/vmerel/keys
#manual -> aspera_RAD.key

#list of file
mkdir /scratch/vmerel/ncbi_tmp_RAD
mkdir /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/RAD

scp ~/Desktop/ncbi_tmp/tmp_meta.tsv vmerel@curnagl.dcsr.unil.ch://scratch/vmerel/ncbi_tmp_RAD

#just a verif
for file in `tail -n +2 tmp_meta.tsv | cut -f 12`
do
  #echo $file
  gzip -cd  /work/FAC/FBM/DEE/tschwand/myrmecophilus/RAD/clean/$file | head -n 1
  if [ ! -f /work/FAC/FBM/DEE/tschwand/myrmecophilus/RAD/clean/$file ];      then
    echo "File not found!"
  fi
done

echo '#!/usr/bin/env bash

date

source ~/.bashrc
conda activate /scratch/vmerel/envs/aspera

mkdir /scratch/vmerel/ncbi_tmp_RAD
cd /scratch/vmerel/ncbi_tmp_RAD

for file in `tail -n +2 tmp_meta.tsv | cut -f 12`
do
  cp /work/FAC/FBM/DEE/tschwand/myrmecophilus/RAD/clean/$file ./
done

ascp -i /users/vmerel/keys/aspera_RAD.key -QT -l100m -k1 -d /scratch/vmerel/ncbi_tmp_RAD/ subasp@upload.ncbi.nlm.nih.gov:uploads/merel.vincent_gmail.com_Rtsqk1nS/myrmeco_RAD

date
' > /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/RAD/NCBI_22122024.sh

nohup sh /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/RAD/NCBI_22122024.sh &> /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/RAD/NCBI_22122024.log &
 
#mv log and sh
mv /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/SR/DNA/RAD/* /nas/FAC/FBM/DEE/tschwand/myrmecophilus/LTS/Reads/DNA/RAD/
```


# mtDNA

```{bash, eval=F}
java -jar /home/vmerel/softs/macse_v2.07.jar

java -jar /home/vmerel/softs/macse_v2.07.jar \
        -prog alignSequences \
        -seq /media/vmerel/8985-27AF/all_filtered.fasta \
        -gc_def 5 \
        -out_NT /media/vmerel/8985-27AF/all_filtered.NT_tmp \
        -out_AA /media/vmerel/8985-27AF/all_filtered.AA_tmp  
        
        java -jar /home/vmerel/softs/macse_v2.07.jar \
        -prog exportAlignment \
        -align /media/vmerel/8985-27AF/all_filtered.NT_tmp \
        -codonForInternalStop NNN \
        -codonForInternalFS --- \
        -codonForFinalStop --- \
        -charForRemainingFS N \
        -out_NT /media/vmerel/8985-27AF/all_filtered.NT \
        -out_AA /media/vmerel/8985-27AF/all_filtered.AA 
   
/home/vmerel/softs/seaview/Gblocks /media/vmerel/8985-27AF/all_filtered.NT \
-t=c 

#Manually removing outgroup
#NCBI not happy
#[] CDS_HAS_FRAMESHIFT

#254a
#270a
#271a
#272a
#273a
#71b

#For each seq, is complaining about residue 44 or 48 to 57, will just remove the first 60 residues for this seq ... I don't have any other sol

~/softs/seaview/seaview /media/vmerel/8985-27AF/all_filtered.NT-gb_noOutgroup 



~/softs/seaview/seaview convert \
-gblocks /media/vmerel/8985-27AF/all_filtered.NT \
-o -

awk '!/gryllotalpa*/' RS=">" ORS=">" /media/vmerel/8985-27AF/all_filtered.fst | sed 's/-//g' | seqtk seq -l0 > /media/vmerel/8985-27AF/all_filtered.ncbi
```

```{R, eval=F}
library(stringr)
#From the sup. directly
Samples <- read.delim("~/Papers_git/rnx/Samples.csv")

#168a is missing is mitoch_id
Samples$mitoch_id[Samples$sample=="19.0168a"]="myrmecophilus"


#isolate to status
Samples$status[Samples$status=="femelle"]="Female"
Samples$status[Samples$status=="larve femelle"]="Female larvae"
Samples$status[Samples$status=="larve indet"]="Larvae"
Samples$status[Samples$status=="larve male"]="Male larvae"
Samples$status[Samples$status=="male"]="Male"

Samples$country = ifelse(Samples$`dep.`=="Grece:", "Greece", "France:")

Samples_mitoch <- Samples %>%
  filter(!is.na(mitoch_id))
#Sequence_ID Orgnaism collection_date geo_loc_name isolate
df <- data.frame(
  Sequence_ID = 
    ifelse(grepl("19.000",Samples_mitoch$sample), str_remove(Samples_mitoch$sample, "19.000"),
           ifelse(grepl("19.00",Samples_mitoch$sample), str_remove(Samples_mitoch$sample, "19.00") , str_remove(Samples_mitoch$sample, "19.0"))),
  organism=
    paste("Myrmecophilus",
          ifelse(Samples_mitoch$mitoch_id %in% c("ambiguous","unknown"), "sp.", Samples_mitoch$mitoch_id)),
  collection_date=paste(Samples_mitoch$yyyy,Samples_mitoch$mm,sep="-"),
  geo_loc_name=paste(Samples_mitoch$country,Samples_mitoch$dep.,sep=" "),
  isolate=paste(Samples_mitoch$sample,Samples_mitoch$status,sep="-"))


write.table(df,"~/Desktop/tmp.tsv", quote=F, sep="\t", row.names=F)

```

